### ISSUE TITLE
Tuple size mismatch for TypeVarTuple regression introduced in 1.1.340

### ISSUE THREAD

**Issue by colehaus** [microsoft/pyright#7285]
```python
from __future__ import annotations

from typing import TypeVarTuple

Shape = TypeVarTuple("Shape")

def full(x: tuple[*Shape]):
    ...

def foo(x: tuple[*Shape, int, int]):
    full(x)
```

I would expect this code to type check. Indeed it does with pyright 1.1.339 and mypy 1.8.0. But not in pyright 1.1.340 or 1.1.350.

```sh
➜ npm install pyright@1.1.339
➜ pyright ml_scratch/type.py
0 errors, 0 warnings, 0 informations

➜ npm install pyright@1.1.340
➜ pyright ml_scratch/type.py
ml_scratch/type.py:13:10 - error: Argument of type "tuple[*Shape@foo, int, int]" cannot be assigned to parameter "x" of type "tuple[*Shape@full]" in function "full"
    "tuple[*Shape@foo, int, int]" is incompatible with "tuple[*Shape@full]"
      Tuple size mismatch; expected 1 but received 3 (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations

➜ npm install pyright@1.1.350
➜ pyright ml_scratch/type.py
ml_scratch/type.py:13:10 - error: Argument of type "tuple[*Shape@foo, int, int]" cannot be assigned to parameter "x" of type "tuple[*Shape@full]" in function "full"
    "tuple[*Shape@foo, int, int]" is incompatible with "tuple[*Shape@full]"
      Tuple size mismatch; expected 0 or more but received indeterminate (reportArgumentType)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Thanks for filing the issue. I agree this is a regression. I've further narrowed it down to [this change](https://github.com/microsoft/pyright/commit/779e64564eb2d762545b996fdd65356cf48cf7ae). I'll investigate further.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
